packages/typespec-ts/src/framework/load-static-helpers.ts

```diff
@@ -0,0 +1,173 @@
+import { readdir, stat, readFile } from "fs/promises";
+import * as path from "path";
+import {
+  ClassDeclaration,
+  FunctionDeclaration,
+  InterfaceDeclaration,
+  Project,
+  SourceFile,
+  TypeAliasDeclaration
+} from "ts-morph";
+import { refkey } from "./refkey.js";
+
+const __dirname = path.dirname(new URL(import.meta.url).pathname);
+
+export const SourceFileSymbol = Symbol("SourceFile");
+export interface StaticHelperMetadata {
+  name: string;
+  kind: "function" | "interface" | "typeAlias" | "class";
+  location: string;
+  [SourceFileSymbol]?: SourceFile;
+}
+
+export function isStaticHelperMetadata(
+  metadata: any
+): metadata is StaticHelperMetadata {
+  return Boolean(
+    metadata &&
+      metadata.name &&
+      metadata.kind &&
+      metadata.location &&
+      metadata[SourceFileSymbol]
+  );
+}
+
+export type StaticHelpers = Record<string, StaticHelperMetadata>;
+
+const DEFAULT_STATIC_HELPERS_PATH = path.resolve(
+  __dirname,
+  "../modular/static-helpers"
+);
+
+export interface LoadStaticHelpersOptions {
+  helpersAssetDirectory?: string;
+  sourcesDir?: string;
+}
+
+export async function loadStaticHelpers(
+  project: Project,
+  helpers: StaticHelpers,
+  options: LoadStaticHelpersOptions = {}
+): Promise<Map<string, StaticHelperMetadata>> {
+  const sourcesDir = options.sourcesDir ?? "src";
+  const helpersMap = new Map<string, StaticHelperMetadata>();
+  const files = await traverseDirectory(
+    options.helpersAssetDirectory ?? DEFAULT_STATIC_HELPERS_PATH
+  );
+
+  for (const file of files) {
+    const targetPath = path.resolve(sourcesDir, file.target);
+    const contents = await readFile(file.source, "utf-8");
+    const addedFile = project.createSourceFile(targetPath, contents);
```

Comment:
it seems we could resolve this issue: https://github.com/Azure/autorest.typescript/issues/2391 after this refactoring.